internal/incident/incidents.go

```diff
@@ -132,3 +136,48 @@ func GetCurrentIncidents() map[int64]*Incident {
 	}
 	return m
 }
+
+// ProcessEvent from an event.Event.
+//
+// This function first gets this Event's object.Object and its incident.Incident. Then, after performing some safety
+// checks, it calls the Incident.ProcessEvent method.
+//
+// The returned error might be wrapped around ErrSuperfluousStateChange.
+func ProcessEvent(
+	ctx context.Context,
+	db *icingadb.DB,
+	logs *logging.Logging,
+	runtimeConfig *config.RuntimeConfig,
+	ev *event.Event,
+) error {
+	obj, err := object.FromEvent(ctx, db, ev)
+	if err != nil {
+		return fmt.Errorf("cannot sync event object: %w", err)
+	}
+
+	createIncident := ev.Severity != event.SeverityNone && ev.Severity != event.SeverityOK
+	currentIncident, created, err := GetCurrent(
+		ctx,
+		db,
+		obj,
+		logs.GetChildLogger("incident"),
+		runtimeConfig,
+		createIncident)
+	if err != nil {
+		return fmt.Errorf("cannot get current incident for %q: %w", obj.DisplayName(), err)
+	}
+
+	if currentIncident == nil {
+		switch {
+		case ev.Type == event.TypeAcknowledgement:
+			return fmt.Errorf("%q does not have an active incident, ignoring acknowledgement event from source %d",
+				obj.DisplayName(), ev.SourceId)
+		case ev.Severity != event.SeverityOK:
+			return fmt.Errorf("cannot process event with a non-OK state without a known incident")
```

Comment:
Changed.

I just disliked raising panics for things which are imo not a total catastrophe, as this goes on the cost resilience. However, in this case this would indicate either an rogue or faulty Icinga 2 daemon.

Reply:
> However, in this case this would indicate either an rogue or faulty Icinga 2 daemon.

Can this happen? When I wrote that code, this was like "this should never happen, if `ev.Severity != event.SeverityOK`, there should be an incident (created if it didn't exist so far), add this check with a more helpful message than the nil dereference that would probably happen right afterwards".